astrocut/cutout_processing.py

```diff
@@ -0,0 +1,429 @@
+                     np.stack((lower_y, lower_y + size[1]), axis=1)), axis=1).astype(int)
+
+
+def _combine_bounds(bounds1, bounds2):
+    """
+    Given two bounds of the form [[x_min, x_max],[y_min, y_max]],
+    combine them into a new [[x_min, x_max],[y_min, y_max]], that
+    encompasses both initial bounds.
+    """
+    
+    bounds_comb = np.zeros((2, 2), dtype=int)
+    bounds_comb[0, 0] = bounds1[0, 0] if (bounds1[0, 0] < bounds2[0, 0]) else bounds2[0, 0]
+    bounds_comb[1, 0] = bounds1[1, 0] if (bounds1[1, 0] < bounds2[1, 0]) else bounds2[1, 0]
+    bounds_comb[0, 1] = bounds1[0, 1] if (bounds1[0, 1] > bounds2[0, 1]) else bounds2[0, 1]
+    bounds_comb[1, 1] = bounds1[1, 1] if (bounds1[1, 1] > bounds2[1, 1]) else bounds2[1, 1]
+    
+    return bounds_comb
+
+
+def _area(bounds):
+    """
+    Given bounds of the form [[x_min, x_max],[y_min, y_max]] return
+    the area of the described rectangle.
+    """
+    return (bounds[0, 1] - bounds[0, 0]) * (bounds[1, 1] - bounds[1, 0])
+
+
+def _get_args(bounds, img_wcs):
+    """
+    Given bounds of the form [[x_min, x_max],[y_min, y_max]] and a
+    `~astropy.wcs.WCS` object return a center coordinate and size
+    ([ny, nx] pixels) suitable for creating a rectangular cutout.
+    """
+    nx = bounds[0, 1]-bounds[0, 0]
+    ny = bounds[1, 1]-bounds[1, 0]
+    x = nx/2 + bounds[0, 0]
+    y = ny/2 + bounds[1, 0]
+    return {"coordinates": img_wcs.pixel_to_world(x, y),
+            "size": (ny, nx)}
+
+
+# TODO: Put this in utils 
+def path_to_footprints(path, size, img_wcs, max_pixels=10000):
+    """
+    Given a path that intersects with a wcs footprint, return
+    one or more rectangles that fully contain that intersection 
+    (plus padding given by 'size') with each rectangle no more than 
+    max_pixels in size.
+    
+    Parameters
+    ----------
+    path : `~astropy.coordinate.SkyCoord`
+        SkyCoord object list of coordinates that represent a continuous path.
+    size : array
+        Size of the rectangle centered on the path locations that must 
+        be included in the returned footprint(s). Formatted as [ny,nx]
+    img_wcs : `~astropy.wcs.WCS`
+        WCS object the path intersects with. Must include naxis information.
+    max_pixels : int
+        Optional, default 10000. The maximum area in pixels for individual
+        footprints.
+        
+    Returns
+    -------
+    response : list
+       List of footprints, each a dictionary of the form:
+       {'center_coord': `~astropy.coordinate.SkyCoord`, 'size': [ny,nx]}
+    """
+    
+    x, y = img_wcs.world_to_pixel(path)
+    
+    # Removing any coordinates outside of the img wcs
+    valid_locs = ((x >= 0) & (x < img_wcs.array_shape[0])) & ((y >= 0) & (y < img_wcs.array_shape[1]))
+    x = x[valid_locs]
+    y = y[valid_locs]
+    
+    bounds_list = _get_bounds(x, y, size)
+    
+    combined_bounds = []
+    cur_bounds = None
+    for bounds in bounds_list:
```

Comment:
This for loop took me quite a while to parse. Maybe some comments throughout it would help? 

I wonder: to avoid the if statement could you perhaps do something like:
```
cur_bounds = bounds_list[0]
for bounds in bounds_list[1:]:
    new_bounds = _combine_bounds(cur_bounds, bounds)
    etc etc
```